crates/common/Cargo.toml

```diff
@@ -7,7 +7,8 @@ edition = { workspace = true }
 [dependencies]
 anyhow = "1.0"
 sha2 = "0.10"
+thiserror = "1.0.40"
```

Comment:
nit: we don't seem to have a strong policy on versioning but is there any reason not to do `thiserror = "1.0"`?

Reply:
I found https://users.rust-lang.org/t/psa-please-specify-precise-dependency-versions-in-cargo-toml/71277 pretty convincing, plus `cargo add` and `cargo update` do this by default, so I've been following that practice by default.

Reply:
IMO it doesn't really matter for binary crates, but I'd agree that precise is better for libs.

Reply:
I don't have strong opinions here. The resource you posted is helpful (I've not seen it before), but it more strongly argues for avoiding just having the major version number which I agree with. If the crate is following semver, then not constraining the patch number should be fine.

Reply:
This is no longer relevant since I've reverted the commit that added the dependency.